File: java/decorator/StarbuzzCoffee/decorator/condiments/Milk.java
package decorator.condiments;

import component.Beverage;
import decorator.CondimentDecorator;

public class Milk extends CondimentDecorator {

    private Beverage beverage;

    public Milk(Beverage beverage) { this.beverage = beverage; }

    @Override
    public double cost() { return beverage.cost() + 0.20; }

    @Override
    public String getDescription() { return beverage.getDescription() + ", Milk"; }
}

File: kotlin/decorator/StarbuzzCoffee/decorator/condiments/Milk.kt
package decorator.condiments

import component.Beverage
import decorator.CondimentDecorator

class Milk(beverage: Beverage) : CondimentDecorator() {

    private var beverage: Beverage? = null

    init { this.beverage = beverage }

    override fun getDescription(): String { return beverage!!.getDescription() + ", Milk" }

    override fun cost(): Double { return beverage!!.cost() + 0.89 }

}